addons/account/tests/test_account_payment.py

```diff
@@ -684,3 +684,17 @@ def assert_paid_hook_call(subtest_msg):
             create_statement_line_and_reconcile(payment=payment, amount=invoice.amount_total / 2)
             payment = register_payment(invoice, line_without_outstanding, invoice.amount_total / 2)
             create_statement_line_and_reconcile(invoice=invoice, amount=invoice.amount_total / 2)
+
+    def test_vendor_payment_save_user_selected_journal_id(self):
+        journal_bank = self.env['account.journal'].search([('name', '=', 'Bank')])
+        journal_cash = self.env['account.journal'].search([('name', '=', 'Cash')])
+
+        self.partner.property_outbound_payment_method_line_id = journal_cash.outbound_payment_method_line_ids
+
+        with Form(self.env['account.payment'].with_context(default_payment_type='outbound')) as payment:
```

Comment:
You do not need to trigger the onchange since your modification is in a compute :thinking: